sos/plugins/krb5.py

```diff
@@ -19,8 +19,34 @@ class Krb5(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
     packages = ('krb5-libs', 'krb5-user')
 
     def setup(self):
-        self.add_copy_spec("/etc/krb5.conf")
+        self.add_copy_spec([
+            "/etc/krb5.conf",
+            "/etc/krb5.conf.d/*",
+            "/var/log/krb5kdc.log",
+            "/var/log/kadmind.log",
+        ])
         self.add_cmd_output("klist -ket /etc/krb5.keytab")
 
 
+class RedHatKrb5(Krb5, RedHatPlugin):
+
+    def setup(self):
+        super(RedHatKrb5, self).setup()
+        self.add_copy_spec([
+            "/var/kerberos/krb5kdc/kadm5.acl",
+            "/var/kerberos/krb5kdc/kdc.conf",
+        ])
+        self.add_cmd_output("klist -ekt /var/kerberos/krb5kdc/.k5*")
```

Comment:
Since the only difference between `RedHatKrb5` and `DebianKrb5` is the directory in which these files are located, I think it would be better to move these collections to the base `Krb5.setup()`, and have the tagging classes set the directory instead of specifying them individually for each distro.  E.G. something like:

```python 
class Krb5(Plugin):

    kdir = "/whatever/upstream/default/is"

    def setup(self):
    [...]
    kfiles = ["kadm5.acl", "kdc.conf"]
    self.add_copy_spec([
        "%s/%s" % (self.kdir, kfile) for kfile in kfiles
    ])

class RedHatKrb5(Krb5, RedHatPlugin):

    def setup(self):
        self.kdir = "/var/kerberos/krb5kdc"
        super(RedHatKrb5, self).setup()
```

This would make adding both more distro support and more files to collect easier down the road.

Reply:
Ack

Reply:
(ditto for the `klist`s)

Reply:
This makes sense.  It's a bit more complicated than that because neither actually use the upstream default (/var/krb5kdc), but I'll see what I can do.

Reply:
> This makes sense. It's a bit more complicated than that because neither actually use the upstream default (/var/krb5kdc), but I'll see what I can do.

Yeah: that's a bit of a weird case. One way to do it that reflects reality at least is to make the upstream default the value in the base class (which is effectively an abstract `Plugin`), and then override it in both children... so the upstream one is never actually really used... It's more documentation than anything at that point, but it's one option.

Reply:
(the alternative is to pick one of the real ones for the base class and just add a comment explaining it).